### PR TITLE
feat: updating Tensorboard related code to use v1

### PR DESCRIPTION
**Pull request by yfang1**
Fallout of #849

### CODE REVIEW

File: google/cloud/aiplatform/tensorboard/plugins/tf_profiler/profile_uploader.py [hunk 1]
```diff
@@ -38,20 +38,18 @@
 import tensorflow as tf
 
 from google.cloud import storage
-from google.cloud.aiplatform.compat.services import tensorboard_service_client_v1beta1
+from google.cloud.aiplatform.compat.services import tensorboard_service_client_v1
+from google.cloud.aiplatform.compat.types import tensorboard_data_v1 as tensorboard_data
```
Comment (morgandu) [inline google/cloud/aiplatform/tensorboard/plugins/tf_profiler/profile_uploader.py:42]: We append the `*_v1beta1` i.e. `tensorboard_service_client_v1beta1` only when the `v1beta1` version is in use which is against compat's `DEFAULT_VERSION = V1`. 

For all `compat` supporting `types` and `services`, you can go with i.e 
`from google.cloud.aiplatform.compat.types import tensorboard_data` where `types.tensorboard_data = types.tensorboard_data_v1` see config here: https://github.com/googleapis/python-aiplatform/blob/main/google/cloud/aiplatform/compat/__init__.py#L24

The lines below should be sufficient for calling v1 as the default version:
```
from google.cloud.aiplatform.compat.services import tensorboard_service_client
from google.cloud.aiplatform.compat.types import tensorboard_data
```

**Review by morgandu — CHANGES_REQUESTED** [review]

**Comment by morgandu** [comment]
Left a comment which applies to all the changes. So the rule of thumb is: 

if your expected version aligns with the `DEFAULT_VERSION` in the `compact` module, no need to specify version i.e. `from google.cloud.aiplatform.compat.types import tensorboard_data`; but if your expected version is different than the `DEFAULT_VERSION`, import the particular version. i.e. `from google.cloud.aiplatform.compat.types import tensorboard_data_v1beta1`.

File: google/cloud/aiplatform/tensorboard/plugins/tf_profiler/profile_uploader.py (hunk 1 above)
Comment (yfang1) [reply google/cloud/aiplatform/tensorboard/plugins/tf_profiler/profile_uploader.py]: ok changed. Thanks!

File: google/cloud/aiplatform/tensorboard/uploader_utils.py [hunk 2]
```diff
@@ -201,7 +195,7 @@ def _create_or_get_run_resource(
                 Required. The display name of this run.
 
         Returns:
-            tb_run (google.cloud.aiplatform_v1beta1.types.TensorboardRun):
+            tb_run (google.cloud.aiplatform_v1.types.TensorboardRun):
```
Comment (morgandu) [inline google/cloud/aiplatform/tensorboard/uploader_utils.py:198]: ```tb_run (tensorboard_run.TensorboardRun):```

File: tests/unit/aiplatform/test_uploader.py [hunk 3]
```diff
@@ -46,22 +46,18 @@
 from google.cloud.aiplatform.tensorboard.plugins.tf_profiler import profile_uploader
 import google.cloud.aiplatform.tensorboard.uploader as uploader_lib
 from google.cloud import storage
-from google.cloud.aiplatform.compat.services import tensorboard_service_client_v1beta1
-from google.cloud.aiplatform_v1beta1.services.tensorboard_service.transports import (
+from google.cloud.aiplatform.compat.services import tensorboard_service_client
```
Comment (morgandu) [inline tests/unit/aiplatform/test_uploader.py:49]: ```from google.cloud.aiplatform_v1.services import tensorboard_service_client```

File: tests/unit/aiplatform/test_uploader.py [hunk 4]
```diff
@@ -46,22 +46,18 @@
...
+from google.cloud.aiplatform_v1.services.tensorboard_service.transports import (
     grpc as transports_grpc,
 )
+from google.cloud.aiplatform.compat.types import tensorboard_data
+from google.cloud.aiplatform.compat.types import tensorboard_service
```
Comment (morgandu) [inline tests/unit/aiplatform/test_uploader.py:54]:
```
from google.cloud.aiplatform_v1.types import tensorboard_data
from google.cloud.aiplatform_v1.types import tensorboard_service
```

File: tests/unit/aiplatform/test_uploader.py [hunk 5]
```diff
@@ -46,22 +46,18 @@
...
+from google.cloud.aiplatform.compat.types import tensorboard_run as tensorboard_run_type
 from google.cloud.aiplatform.compat.types import (
-    tensorboard_experiment_v1beta1 as tensorboard_experiment_type,
-)
-from google.cloud.aiplatform.compat.types import (
-    tensorboard_run_v1beta1 as tensorboard_run_type,
-)
-from google.cloud.aiplatform.compat.types import (
-    tensorboard_time_series_v1beta1 as tensorboard_time_series_type,
+    tensorboard_time_series as tensorboard_time_series_type,
```
Comment (morgandu) [inline tests/unit/aiplatform/test_uploader.py:60]: dito

File: tests/unit/aiplatform/test_uploader.py [hunk 6]
```diff
@@ -46,22 +46,18 @@
...
     grpc as transports_grpc,
 )
+from google.cloud.aiplatform.compat.types import tensorboard_data
+from google.cloud.aiplatform.compat.types import tensorboard_service
 from google.cloud.aiplatform.compat.types import (
-    tensorboard_data_v1beta1 as tensorboard_data,
-    tensorboard_service_v1beta1 as tensorboard_service,
+    tensorboard_experiment as tensorboard_experiment_type,
 )
+from google.cloud.aiplatform.compat.types import tensorboard_run as tensorboard_run_type
```
Comment (morgandu) [inline tests/unit/aiplatform/test_uploader.py:58]: dito

File: tests/unit/aiplatform/test_uploader.py [hunk 7]
```diff
@@ -46,22 +46,18 @@
...
+from google.cloud.aiplatform.compat.services import tensorboard_service_client
+from google.cloud.aiplatform_v1.services.tensorboard_service.transports import (
     grpc as transports_grpc,
 )
+from google.cloud.aiplatform.compat.types import tensorboard_data
+from google.cloud.aiplatform.compat.types import tensorboard_service
 from google.cloud.aiplatform.compat.types import (
-    tensorboard_data_v1beta1 as tensorboard_data,
-    tensorboard_service_v1beta1 as tensorboard_service,
+    tensorboard_experiment as tensorboard_experiment_type,
```
Comment (morgandu) [inline tests/unit/aiplatform/test_uploader.py:56]: dito

**Review by morgandu — APPROVED** [review]
Sorry I probably missed the `test_uploader.py` in the previous review. For unit tests, we don't rely on the `compat` module, instead we specify the expected version. Because we don't have tests when `compat` may or may not select different version within different resource class methods, this way we can ensure class methods calling `compat` returning the right version we expect.

We only discussed this when `compat` was introduced, but did not enforce the implementation across all other module, so there is some inconsistency in the code base. But prefer start to enforce it for the reason mentioned above.

**Pull request merged**